modules/configure_project/main.tf

```diff
@@ -0,0 +1,102 @@
+                      "resource_crn": "${var.cos_crn}",
+                      "delegated": ${var.watsonx_project_delegated}
+                    },
+                    "description": "${var.watsonx_project_description}",
+                    "public": true,
+                    "tags": ${jsonencode(var.watsonx_project_tags)},
+                    "compute": [
+                      {
+                        "name": "${var.machine_learning_name}",
+                        "guid": "${var.machine_learning_guid}",
+                        "type": "machine_learning",
+                        "crn": "${var.machine_learning_crn}"
+                      }
+                    ],
+                    "settings": {
+                      "access_restrictions": {
+                        "data": ${var.watsonx_mark_as_sensitive}
+                      }
+                    }
+                  }
+                  EOT
+  update_method  = "PATCH"
+  update_path    = "${local.dataplatform_api}{id}"
+  update_data    = <<-EOT
+                  {
+                    "name": "${var.watsonx_project_name}",
+                    "type": "wx",
+                    "description": "${var.watsonx_project_description}",
+                    "public": true,
+                    "compute": [
+                      {
+                        "name": "${var.machine_learning_name}",
+                        "guid": "${var.machine_learning_guid}",
+                        "type": "machine_learning",
+                        "crn": "${var.machine_learning_crn}",
+                        "credentials": {}
+                      }
+                    ]
+                  }
+                  EOT
+}
+
+resource "time_sleep" "wait_5_seconds" {
+  depends_on      = [restapi_object.configure_project]
+  create_duration = "5s"
+}
+
+data "restapi_object" "get_project" {
+  depends_on   = [resource.restapi_object.configure_project, resource.time_sleep.wait_5_seconds]
+  path         = "${local.dataplatform_api}/v2/projects"
+  query_string = "project_ids=${local.watsonx_project_id}"
+  results_key  = "resources"
+  search_key   = "metadata/guid"
+  search_value = local.watsonx_project_id
+  id_attribute = "metadata/guid"
+}
+
+locals {
+  dataplatform_api_mapping = {
+    "us-south" = "//api.dataplatform.cloud.ibm.com",
+    "eu-gb"    = "//api.eu-gb.dataplatform.cloud.ibm.com",
+    "eu-de"    = "//api.eu-de.dataplatform.cloud.ibm.com",
+    "jp-tok"   = "//api.jp-tok.dataplatform.cloud.ibm.com"
+  }
+  dataplatform_api          = local.dataplatform_api_mapping[var.region]
+  watsonx_project_id_object = restapi_object.configure_project.id
+  watsonx_project_id        = regex("^.+/([a-f0-9\\-]+)$", local.watsonx_project_id_object)[0]
+  watsonx_project_data      = jsondecode(data.restapi_object.get_project.api_response)
+  dataplatform_ui_mapping = {
+    "us-south" = "https://dataplatform.cloud.ibm.com",
+    "eu-gb"    = "https://eu-uk.dataplatform.cloud.ibm.com",
```

Comment:
Shouldn't this be `eu-gb.dataplatform...` instead of `eu-uk`?

Reply:
yes, good catch. Its a typo mistake.